pkg/html/html.go

```diff
@@ -727,6 +731,188 @@ func testImpactingComponents(testImpactingBugs []bugsv1.Bug) string {
+			</td>
+			<td>
+				%0.2f%% <span class="text-nowrap">(%d runs)</span>
+			</td>
+			<td>
+				%s
+			</td>
+			<td>
+				%0.2f%% <span class="text-nowrap">(%d runs)</span>
+			</td>
+		</tr>
+	`
+
+	naTemplate := `
+			<tr class="%s">
+				<td>
+					%[2]s
+					<p>
+					<button class="btn btn-primary btn-sm py-0" style="font-size: 0.8em" type="button" data-toggle="collapse" data-target=".%[3]s" aria-expanded="false" aria-controls="%[3]s">Expand Failing Jobs</button>
+				</td>
+				<td>
+					%0.2f%% <span class="text-nowrap">(%d runs)</span>
+				</td>
+				<td/>
+				<td>
+					NA
+				</td>
+			</tr>
+		`
+
+	bzFailingJobTemplate := `
+			<tr class="collapse %s">
+				<td>
+					%[2]s -- %[3]s
+					<p>
+					<button class="btn btn-primary btn-sm py-0" style="font-size: 0.8em" type="button" data-toggle="collapse" data-target=".%[4]s" aria-expanded="false" aria-controls="%[4]s">Expand Failing Tests</button>
+				</td>
+				<td>
+					%0.2f%% <span class="text-nowrap">(%d/%d failed/total)</span>
+				</td>
+				<td/>
+				<td>
+					NA
+				</td>
+			</tr>
+		`
+
+	for _, v := range failuresByBugzillaComponent {
+		safeBZJob := fmt.Sprintf("%s---component", v.Name)
+		safeBZJob = strings.ReplaceAll(safeBZJob, ".", "")
+		safeBZJob = strings.ReplaceAll(safeBZJob, " ", "")
+
+		prev := util.GetPrevBugzillaJobFailures(v.Name, failuresByBugzillaComponentPrev)
+		highestFailPercentage := v.JobsFailed[0].FailPercentage
+		lowestPassPercentage := 100 - highestFailPercentage
+		rowColor := ""
+		switch {
+		case lowestPassPercentage > 85:
+			rowColor = "table-success"
+		case lowestPassPercentage > 70:
+			rowColor = "table-warning"
+		case lowestPassPercentage > 0:
+			rowColor = "table-danger"
+		default:
+			rowColor = "error"
+		}
+
+		if prev != nil && len(prev.JobsFailed) > 0 {
+			previousHighestFailPercentage := prev.JobsFailed[0].FailPercentage
+			previousLowestPassPercentage := 100 - previousHighestFailPercentage
+			arrow := ""
+
+			delta := 5.0
+			if v.JobsFailed[0].TotalRuns > 80 {
+				delta = 2
+			}
+			if lowestPassPercentage > previousLowestPassPercentage+delta {
+				arrow = up
+			} else if lowestPassPercentage < previousLowestPassPercentage-delta {
+				arrow = down
+			}
+
+			if lowestPassPercentage > previousLowestPassPercentage+delta {
+				arrow = fmt.Sprintf(up, lowestPassPercentage-previousLowestPassPercentage)
+			} else if lowestPassPercentage < previousLowestPassPercentage-delta {
+				arrow = fmt.Sprintf(down, previousLowestPassPercentage-lowestPassPercentage)
+			} else if lowestPassPercentage > previousLowestPassPercentage {
+				arrow = fmt.Sprintf(flatup, lowestPassPercentage-previousLowestPassPercentage)
+			} else {
+				arrow = fmt.Sprintf(flatdown, previousLowestPassPercentage-lowestPassPercentage)
+			}
+
+			s = s + fmt.Sprintf(bzGroupTemplate,
+				rowColor,
+				v.Name,
+				safeBZJob,
+				lowestPassPercentage,
+				prev.JobsFailed[0].TotalRuns,
```

Comment:
something seems off about this value.  i'm seeing pretty low values reported.  if it were the number of failed runs the numbers would make sense, but as a "total" they don't make sense to me.  let's discuss when you're free.

or maybe it's low because you just picked the first entry?  should this be a sum of all the entries?

Reply:
also i don't thiink you mean to have `prev` here.  should be the current week